### PR TITLE
export FT_Outline_Decompose and tests

### PR DESCRIPTION
**Pull request by wookay**
rebase from #22

### CODE REVIEW

**Comment by Gnimuc** [comment]
Shall we export all of the `FT_` prefixed APIs?

```julia
foreach(names(@__MODULE__, all=true)) do s
    if startswith(string(s), "FT_")
        @eval export $s
    end
end
```

**Comment by wookay** [comment]
there are 635 functions, constants, types for having `FT_`. I think that's so massive.
and normally `export` used at the top of the module.
```
julia> module G
       FT_A=2
       foreach(names(@__MODULE__, all=true)) do s
           if startswith(string(s), "FT_")
               @eval export $s
           end
       end
       FT_B=3
       end
Main.G

julia> using .G

julia> FT_A
2

julia> FT_B
ERROR: UndefVarError: FT_B not defined
```

**Comment by Gnimuc** [comment]
I'm currently using a pattern that every wrapper package should contain two modules:
1. a submodule `LibXXX.jl` which is one-to-one mapped to the C API with everything being exported.
2. a module `XXX.jl` built upon `LibXXX.jl`, where only useful helper functions are exported.

When translating code from C to Julia, people may prefer to use `using LibXXX.jl`, but when building Julia projects from scratch, `using XXX.jl` and its high-level interfaces are basically enough, not to mention that users could also `import LibXXX: yyy` even if `yyy` is missing in the high-level APIs.

File: test/runtests.jl [hunk 1]
```diff
@@ -1,9 +1,61 @@
...
+
+move_f = @cfunction $move_to_func Cint (Ptr{FT_Vector}, Ptr{Cvoid})
+line_f = @cfunction $line_to_func Cint (Ptr{FT_Vector}, Ptr{Cvoid})
+conic_f = @cfunction $conic_to_func Cint (Ptr{FT_Vector}, Ptr{FT_Vector}, Ptr{Cvoid})
+cubic_f = @cfunction $cubic_to_func Cint (Ptr{FT_Vector}, Ptr{FT_Vector}, Ptr{FT_Vector}, Ptr{Cvoid})
+
+face = unsafe_load(refface[])
+glyph = unsafe_load(face.glyph)
+outline_funcs = FreeType.FT_Outline_Funcs(Base.unsafe_convert.(Ptr{Cvoid}, (move_f, line_f, conic_f, cubic_f))..., 0, 0)
+FT_Outline_Decompose(pointer_from_objref.((Ref(glyph.outline), Ref(outline_funcs)))..., C_NULL)
```
Comment (SimonDanisch) [inline test/runtests.jl:52]: that's unsafe and might explain the segfault

**Comment by SimonDanisch** [comment]
Thank you :)

**Comment by wookay** [comment]
@Gnimuc  probably file an issue about the exporting all `FT_`s, by necessity.    I really appreciated all of your work, indeed. I'm heavily using `Nuklear.jl` these days.

@SimonDanisch you're welcome!

**Comment by Gnimuc** [comment]
@wookay I'm going to regenerate FreeType C-bindings using latest Clang.jl. For compatibility's sake, I'll keep current exported APIs untouched. As for Nuklear.jl, I feel like we need to add more [helper C functions](https://github.com/Gnimuc/NuklearBuilder/blob/master/wrapper/helper.c) to make it easier to work with. Any feedback is highly appreciated!

**Pull request merged**